### ISSUE TITLE
New contribution document Contents

### ISSUE THREAD

**Issue by rkooo567** [ucbrise/clipper#704]
I am planning to write a new contribution documentation `CONTRIBUTING.md`. Based on this, I will update the webpage's contribution doc. 

I am planning to write these things.

1. The current content (like how to compile query frontend) + custom docker image
2. How our current CI works
3. Link to the design doc, concepts doc, research paper
4. How to work on clipper admin
5. Useful information (ex, https://github.com/ucbrise/clipper/issues/683) -> 683 should not be in contribution doc. 
6. How to deploy KubernetesContainerManager (based on https://github.com/ucbrise/clipper/issues/598). The contents can be changed.

Please recommend me more contents!

**Comment by jacekwachowiak** [comment]
From my experience more on the logging process - how could we understand how it's being sent, how to redirect it somewhere else or use another tool to gather the logging data and so on would be useful.

**Comment by rkooo567** [comment]
@jacekwachowiak For sure! That will be a very important part. I will definitely add it.

**Comment by jacekwachowiak** [comment]
One more thing - I was testing Clipper with different frameworks and so I have some basic working examples on most of them: PySpark, PyTorch, MXNet, Tensorflow, Scikit, XGBoost in form of easy Jupyter notebooks. Do you think it's worth to add them to the repository so that a user can check Clipper deployment without really writing any code and then go from it to more specific solutions? I know the documentation has some code snippets and there is a pair of examples for python and metrics but it's not complete and the documentation may be confusing (it was in case of MXNet and Tensorflow for me). If so, I would create a PR as soon as I clean up the notebooks I have. For some frameworks (TF, Pyspark) I also have a K8s version.

**Comment by rkooo567** [comment]
That actually sounds amazing! We will appreciate any form of contribution to the project. You can write them under `example` folder. Feel free to create new directories if there's no proper category for your tutorial!

**Comment by rkooo567** [comment]
@jacekwachowiak Thanks again for awesome work!

**Comment by rkooo567** [comment]
This is my basic structure.

* [Quick Start](#quick-start)
* [Guideline](#guideline)
    * [Feature Request](#feature-request)
    * [Pull Request](#pull-request)
    * [How to Answer Issues](#how-to-answer-issues)
* [How to Start](#how-to-start)
* [Style Guides](#style-guide)
    * [C++](#cpp)
    * [Python](#python)
* [Development](#development)
    * [CI](#CI)
    * [Client Library Develpment (clipper-admin)](#client-library)
        * [Debug](#debug)
        * [Logging](#logging)
    * [Core Development](#core-development)
* [Want to learn Clipper](#want-to-learn-clipper)
    * [Simple Explanation](#simple-explanation)
        * [Concepts](#concepts)
        * [Architecture](#architecture)
    * [Resources](#resources)

**Comment by jacekwachowiak** [comment]
I was wondering if some hints or good practices about Clipper deployment on different versions of Kubernetes - GKE, EKS. bare metal and so on could be included. It can save a lot of time, if you're sure that your cluster is ready for Clipper. I saw quite a few issues about Clipper never initializing correctly on Google's Kuberenetes or AWS, taking a long time or working only partially and I think it's a safe assumption that a big part of clusters come from those providers. (Yesterday I tried briefly to start Clipper on GKE assuming the out-of-the-box approach - default cluster, default python, default Clipper setup and no luck). For now if I am not wrong, the documentation says a K8s cluster must be provided but past experiences show that things like the proxy, RBAC, permissions matter a lot.

**Comment by rkooo567** [comment]
@jacekwachowiak Good point. I agree with you. Seamless integration with those public cloud services and minikube will be one of our primary goals in the future releases. For now, it is a little hard to write docs for all of different versions of Kubernetes because we don't have sufficient examples yet. Maybe the initial contribution doc wouldn't contain them, but we should definitely add them as time goes. Thanks for the great feedback.
